addons/explosives/functions/fnc_setupExplosive.sqf

```diff
@@ -141,13 +141,6 @@ GVAR(TweakedAngle) = 0;
     //Don't allow placing in a bad position:
     if (_badPosition && {GVAR(placeAction) == PLACE_APPROVE}) then {GVAR(placeAction) = PLACE_WAITING;};
 
-    if (((inputAction "zoomTemp") > 0) || //Cancel on RMB, For some reason this works (when held) but AddActionEventHandler doesn't
-            {_unit != ACE_player} ||
-            {!([_unit, objNull, ["isNotSwimming"]] call EFUNC(common,canInteractWith))} ||
-            {!(_magClassname in (magazines _unit))}) then {
-        GVAR(placeAction) = PLACE_CANCEL;
```

Comment:
We don't want to remove this entire if statement, just the RMB part.
All of the other checks need to be run each frame in the PFEH so it will cancel if any of them happen.
e.g. if `canInteractWith` is false, we need to cancel right away instead of waiting for RMB